codebuild/bin/s2n_dynamic_load_test.c

```diff
@@ -63,9 +64,10 @@ int main(int argc, char *argv[])
     /* s2n-tls library can be dynamically loaded and cleaned up safely 
      *
      * We can't use any s2n test macros because this test doesn't get linked to 
-     * s2n during compile-time.
+     * s2n during compile-time. This test is in a loop to make sure that we are
+     * cleaning up pthread keys properly.
      */
-    {
+    for (size_t i = 0; i <= PTHREAD_KEYS_MAX; i++) {
```

Comment:
Should we test higher than PTHREAD_KEYS_MAX? Wouldn't we expect to be able to create exactly PTHREAD_KEYS_MAX keys?

Reply:
It actually starts to error a few keys before hitting PTHREAD_KEYS_MAX so I suspect the number isn't actually accurate. But I can change it just in case.